### PR TITLE
Prevent `RequestNotWellFormed` error for invalid characters in BBOX requests (ref: layer names with spaces, colons or slashes)

### LINKED ISSUE #565: BBOX request for layers with blank spaces or / in name throws `RequestNotWellFormed`
### Checklist

- [X] I've searched through the existing [issues](https://github.com/g3w-suite/g3w-client/issues) and this bug has never been reported before

### Subject of the issue

In case of layer name with blank space and /, if we do a bbox request we receive an error

Screenshot from 2024-02-02 12-52-08

### Steps to reproduce

1. Do a BBOX request 

### Environment

- g3w-admin: 3.7.3
- g3w-client: 3.9.2
- browser: all

### Link to your project

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by volterra79**
Closes: #565

### CODE REVIEW

File: src/app/core/layers/layersstore.js [hunk 1]
```diff
@@ -112,12 +112,14 @@ proto.getLayersDict = function(filter = {}, options = {}) {
...
+  //in case of SELECTED_OR_ALL = true
   if (filter.SELECTED_OR_ALL) {
     let _layers = layers;
+    //check if there are selected layer
     layers = layers.filter(layer => layer.isSelected());
-    layers = layers.length ? layers : _layers;
-  } else if (false === filter.SELECTED_OR_ALL) {
+    //otherwise get all layers
+    layers = layers.length > 0 ? layers : _layers;
+  } else if ('boolean' === typeof filter.SELECTED) { //check if filter.SELECTED is Boolean
```
Comment (Raruto) [inline src/app/core/layers/layersstore.js]: @volterra79 so what are the possible values for `filter.SELECTED`?

Comment (volterra79) [reply src/app/core/layers/layersstore.js]: @Raruto can be undefined or null. It depends if is set or not

Comment (volterra79) [reply src/app/core/layers/layersstore.js]: @Raruto these changes solve also a issue related to add feature to result button

Screenshot from 2024-02-06 12-27-04

File: src/app/core/layers/imagelayer.js [hunk 2]
```diff
@@ -171,8 +171,8 @@ proto.getWmsUrl = function({ type = 'map' } = {}) {
   return this.config.wmsUrl;
 };
 
-proto.getWFSLayerName = function(){
-  return this.getQueryLayerName().replace(/[/\s]/g, '_')
+proto.getWFSLayerName = function() {
+  return this.getQueryLayerName().replace(/\s+/g, '_').replace( ':', '-' );
```
Comment (Raruto) [inline src/app/core/layers/imagelayer.js]:
```suggestion
  return this.getQueryLayerName().replace(/\s/g, '_').replaceAll( ':', '-' );
```

Comment (volterra79) [reply src/app/core/layers/imagelayer.js]: @Raruto i think that /g mean replace all (globally)

File: src/utils/parsers/index.js [hunk 3]
```diff
@@ -161,15 +161,19 @@ const utils = {
...
       let originalName = (wms && layer.isWmsUseLayerIds()) ? layer.getId(): layer.getName();
-      let sanitizeLayerName = wms ? originalName.replace(/[/\s]/g, '') : originalName.replace(/[/\s]/g, '_');
-      sanitizeLayerName = sanitizeLayerName.replace(/(\'+)/, '');
-      sanitizeLayerName = sanitizeLayerName.replace(/(\)+)/, '');
-      sanitizeLayerName = sanitizeLayerName.replace(/(\(+)/, '');
+      let sanitizeLayerName = wms ? originalName.replace(/[/\s]/g, '') : originalName.replace(/\s+/g, '_');
+      sanitizeLayerName     = sanitizeLayerName.replace(/(\'+)/, '');
+      sanitizeLayerName     = sanitizeLayerName.replace(/(\)+)/, '');
+      sanitizeLayerName     = sanitizeLayerName.replace(/(\(+)/, '');
+      sanitizeLayerName     = wms ? sanitizeLayerName : sanitizeLayerName.replace(/\/+/g, '');
```
Comment (Raruto) [inline src/utils/parsers/index.js]: @volterra79 I think that by using the "+" modifier may result in rare cases of collisions:

| ORIGINAL | SANITIZED |
|-------|-------|
| `layername`    | `layername` |
| `layer / name` | `layername` |
| `layer / (name)` | `layername` |

If so, let's try to test them directly in here: https://github.com/g3w-suite/g3w-client/pull/541

File: src/app/core/layers/imagelayer.js (hunk 2 above)
Comment (Raruto) [reply src/app/core/layers/imagelayer.js]: @volterra79 for the ":" character you didn't use a regex, just a simple: `replace( ':', '-' );`

File: src/utils/parsers/index.js (hunk 3 above)
Comment (volterra79) [reply src/utils/parsers/index.js]: @Raruto try rename building with **buildings: wfs (a/z)** and run a query by bbox on this. layer.
Qgis server return as layername on response xml **buildings-_wfs_az**.
It replace : with -, blank space with _,  ( ) and / with empty space

**Comment by Raruto** [comment]
> try rename building with **buildings: wfs (a/z)** and run a query by bbox on this. layer.
> Qgis server return as layername on response xml **buildings-_wfs_az**.
> It replace : with -, blank space with _, ( ) and / with empty space

here's what I tried:

image

**Comment by volterra79** [comment]
> > try rename building with **buildings: wfs (a/z)** and run a query by bbox on this. layer.
> > Qgis server return as layername on response xml **buildings-_wfs_az**.
> > It replace : with -, blank space with _, ( ) and / with empty space
> 
> here's what I tried:
> 
> image

@Raruto get last commit

**Review by Raruto — APPROVED** [review]

**Pull request merged**